api/eid_server/eid_types/sessionid.py

```diff
@@ -1,32 +1,18 @@
-import numpy as np
-from numpy import random
-import binascii
+from eid_server.eid_types.type_check import *
+from eid_server.eid_types.hexbinary import *
 
 
 class SessionID:
 
-    def __init__(self):
-        self.__sessionid = ""
+    def __init__(self, id):
+        self.id = id
 
     @property
     def id(self):
-        i = 16
-        while len(self.__sessionid) < 32:
-            a = random.bytes(i)
-            i += 1
-            self.__sessionid = binascii.b2a_hex(a)
-        return self.__sessionid
+        return self._id
 
     @id.setter
     def id(self, value):
-        if len(value) < 32 or len(value) > 64:
-            return False
-
-        for c in value:
-            if (c not in list(range(ord('0'), ord('9') + 1)) 
-                    + list(range(ord('a'), ord('f') + 1))):
-                return False
-
-        self.__sessionid = value
-        return True
+        check("id", value, HexBinary16)
```

Comment:
I refactored as suggested. For being able to rewind the changes, I pushed a whole new commit. If the changes are approved, I will squash it to Commit 702a6c493437e41deb6a27b1cde03bf53c6f47c4 .